gulp/versioncheck.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var request = require('request');
+var pluginUtils = require('../lib/plugin-utils');
+var supportedMinVersion = 15.7;
+
+module.exports = function (gulp, gutil) {
+
+    function process(server, cb) {
+        try {
+            if (server.serverUrl() === undefined) {
+                var errMessage = 'A server URL is required in your configuration. ';
+                callbackOrThrowError(cb, errMessage);
+            }
+            var versionCheckUrl = pluginUtils.urlBldr(server.serverUrl() + 'status/version').build();
+            request(versionCheckUrl, function (error, response, body) {
+                if (error || response.statusCode > 201) {
+                    callbackOrThrowError(cb, error.message);
+                } else {
+                    var matches = body.match(/<body>(\d+(\.\d)*)/i);
+                    if (matches) {
+                        var versionOnServer = matches[1];
+                        if (versionOnServer < supportedMinVersion) {
+                            var errorMessage = "Supported minimum version on server is " + supportedMinVersion;
+                            callbackOrThrowError(cb, errorMessage);
+                        }
+                    } else {
+                        callbackOrThrowError(cb, "Invalid version check response from server");
```

Comment:
not sure if we need any additional messaging here or not.  The message "Invalid version check response from server" might seem a bit cryptic to the developer using the SDK, but I'm sure they will contact support which is probably what they will need to do.  Have you tested the scenario where they are not connected to the internet?  Do they get this error in that case?  If so, then we should probably change the messaging to include a statement about checking to make sure they can connect to the stage site.  Also, please test this flow against a stage site when not connected to the VPN to make sure we don't get a basic auth challenge.